woocommerce/woocommerce-android
# Handle process death while scanning the barcode

Closes: #9096 


### Description

This PR handles the process death when the scanning is in progress. While scanning the barcode using Google's code scanner library, the scanner library opens the scanning UI in a separate process. Meanwhile, if the WooCommerce app process gets destroyed while the scanning is in progress, we will display a toast message saying that the app was terminated and also track the failure so that we can provide a more seamless user experience if this becomes a more common problem for the users.

### Testing instructions

1. Enable `don't keep activities` options from the developer options menu (Android OS settings -> developer options -> don't keep activities)
2. Open the WooCommerce app
3. Start the scanner either from the Order list or from the Order creation screen.
4. Come back from the scanner screen
5. Notice the WooCommerce app gets restarted
6. Ensure you see a toast message that says the app was terminated and to try the operation again
7. Ensure the `barcode_scanning_failure` event is tracked with `VMKilledWhileScanning` as the reason.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/15da58c8-6eda-46a9-9900-69440011ce38
